src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/TableAccessibility.m

```diff
@@ -130,6 +130,15 @@ - (TableRowAccessibility *)createRowWithIndex:(NSUInteger)index
     if (rowCache == nil) {
         int rowCount = [self accessibilityRowCount];
         rowCache = [[NSMutableDictionary<NSNumber*, id> dictionaryWithCapacity:rowCount] retain];
+        cacheValid = YES;
+    }
+
+    if (!cacheValid) {
+        for (NSNumber *key in [rowCache allKeys]) {
+            [[rowCache objectForKey:key] release];
+            [rowCache removeObjectForKey:key];
+        }
+        cacheValid = YES;
```

Comment:
It seems to me that here it is necessary to release the cache.
```
    [rowCache release];
    rowCache = nil;
```

Reply:
If we are removing "all records from the cache", can't we use [removeAllObjects](https://developer.apple.com/documentation/foundation/nsmutabledictionary/1408955-removeallobjects?language=objc)
Guess that will be more optimized..